python/ray/tune/execution/trial_runner.py

```diff
@@ -397,26 +417,31 @@ def restore_from_dir(self, experiment_dir: Optional[str] = None) -> List[Trial]:
         and the callback states. It will then parse the trial states
         and return them as a list of Trial objects.
         """
-        experiment_dir = experiment_dir or self._local_experiment_path
+        if _use_storage_context():
+            raise NotImplementedError(
+                "Restoration is not fully working. TODO for a follow-up PR."
```

Comment:
This will be fixed in a follow-up after piping the storage context to the trials. Raising this here to make the current functionality gap explicit.